tests/ui/macros/expr_2021_cargo_fix_edition.rs

```diff
@@ -0,0 +1,14 @@
+//@  compile-flags: --edition=2021
+// This test ensures that expr_2021 is not allowed on pre-2021 editions
+#![deny(edition_2024_expr_fragment_specifier)]
+
+macro_rules! m {
+    ($e:expr) => { //~ ERROR: the `expr` fragment specifier will accept more expressions in the 2024 edition.
```

Comment:
It might be good to add another clause here that has an `expr` deeper in the pattern to make sure we properly recurse and all that.

Reply:
Ah make really sense!

Reply:
Done in https://github.com/rust-lang/rust/pull/125627/commits/facc35c75c697a5b4dca3a2a78fdd9817201f19e